packages/react-dom/src/events/ReactDOMEventReplaying.js

```diff
@@ -43,6 +43,12 @@ export function setAttemptUserBlockingHydration(fn: (fiber: Object) => void) {
   attemptUserBlockingHydration = fn;
 }
 
+let attemptContinuousHydration: (fiber: Object) => void;
+
+export function setAttemptContinuousHydration(fn: (fiber: Object) => void) {
```

Comment:
It is not obvious to me why we're doing this indirect injection from `ReactDOM` -> `ReactDOMEventReplaying` but it looks like this is copying pre-existing behavior for `setAttemptSynchronousHydration` so I assume it's okay.

Reply:
There's a cycle introduced by event replaying from the reconciler -> host config -> event system -> reconciler. It leads to a mess. I have some ideas for how we can avoid these dynamic injections but it involves getting rid of some other dynamic cases like these need to move into in the functions instead of overriding the functions values during module initialization: https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberCompleteWork.js#L148